alf/networks/networks.py

```diff
@@ -115,3 +118,122 @@ def __init__(self, block, input_tensor_spec=None, activation=torch.relu_):
+
+
+class TemporalStack(Network):
+    """Stack features temporally.
+
+    Suppose input_size=(), stack_size=2, pooling_size=2, the following table
+    shows the output of different mode for an input sequence of 1,2,3,4,5 (ignoring
+    batch dimension)
+
+           1,        2,        3,        4,          5
+    skip:  [0, 1],   [0, 1],   [1, 3],   [1, 3],     [3, 5]
+    avg:   [0, 0],   [0, 1.5], [0, 1.5], [1.5, 3.5], [1.5, 3.5]
+    max:   [0, 0],   [0, 2],   [0, 2],   [2, 4],     [2, 4]
+
+    Example:
+
+    .. code-block:: python
+
+        # A temporal CNN with progressively large temporal receptive field.
+        cnn = alf.networks.Sequential([
+            alf.networks.TemporalStack(256, 3, 1),
+            torch.nn.Flatten(),
+            alf.layers.FC(768, 256, activation=torch.relu_)
+            alf.networks.TemporalStack(256, 3, 2),
+            torch.nn.Flatten(),
+            alf.layers.FC(768, 256, activation=torch.relu_)
+            alf.networks.TemporalStack(256, 3, 4),
+            torch.nn.Flatten(),
+            alf.layers.FC(768, 256, activation=torch.relu_)])
+
+    """
+
+    def __init__(self,
+                 input_size,
+                 stack_size,
+                 pooling_size=1,
+                 dtype=torch.float32,
+                 mode='skip'):
+        """
+        Args:
+            input_size (int|tuple[int]): shape of the input
+            stack_size (int): stack the features from so many steps
+            pooling_size (int): if > 1, perform a pooling first. ``pooling_size``
+                steps of features will be pooled as single feature vector according
+                to ``mode``
+            mode (str): one of ('skip', 'avg', 'max'), only effective if pooling_size > 1.
+                'skip': only keeping features at step ``t * pooling_size``
+                'avg': features are averaged for each window of ``pooling_size`` steps.
+                    The pooling results for first ``pooling_size - 1`` steps are 0.
+                'max': features are maxed for each window of ``pooling_size`` steps
+                    The pooling results for first ``pooling_size - 1`` steps are 0.
+        Returns:
+            tuple of:
+            - tensor of shape (stack_size, input_size)
+            - internal states
+        """
+        if isinstance(input_size, typing.Iterable):
+            input_size = tuple(input_size)
+        else:
+            input_size = (input_size, )
+        shape = (stack_size, ) + input_size
+        input_tensor_spec = alf.TensorSpec(input_size, dtype=dtype)
+        super().__init__(input_tensor_spec)
+        self._pooling_size = pooling_size
+        if pooling_size == 1:
+            self._state_spec = alf.TensorSpec((stack_size - 1, ) + input_size,
+                                              dtype)
+        elif mode == 'skip':
+            self._pool_func = self._skip_pool
+            pool_state_spec = ()
+            self._update_step = 1
```

Comment:
What is the reason for setting ``_update_step`` for 'skip' to be different from the other modes?
This might be useful for keeping information from the first element, but its computational logic is a bit different from others. And it seems this is related to the question raised on the illustrative example as well.
We can at least explain this somewhere.